### PR TITLE
[new unit&bullet] Fallen Angel and Assassin with their special bullet

### PR DESCRIPTION
**Pull request by su040202**
create a new tank "Fallen Angel", and according to my test it works well.

### CODE REVIEW

**Comment by su040202** [comment]
I wish my tank can be merged to main.

**Comment by su040202** [comment]
add new unit Assassin to solve #406 when solve #407 the bullet of it may have a change.

**Comment by su040202** [comment]
add two bullet for the two units and optimize them make them more completed.

**Comment by su040202** [comment]
the bullet hand sword is finished by @zhang-br22, the another part is finished by @su040202.

**Comment by su040202** [comment]
#405

File: src/battle_game/core/units/fallen_angel.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  void TheBlessOfGod();
+  void TheSalvationFromStarsClick();
+  [[nodiscard]] float GetDamageScale() const override;
+  [[nodiscard]] const char *UnitName() const override;
+  [[nodiscard]] const char *Author() const override;
+
+  // float turret_rotation_{0.0f};
+  uint32_t fire_count_down_{0};
+  uint32_t ThePowerFromAbyss_count_down_{300};
+  uint32_t TheSalvationFromStars_count_down_{900};
```
Comment (qinchuanhui) [inline src/battle_game/core/units/fallen_angel.h:27]: Naming problem, shoud look like `fire_count_down_`

File: src/battle_game/core/units/assassin_syl.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  void Invincible();
+  void ChibakuTenseiClick();
+  void ChibakuTensei();
+  [[nodiscard]] float GetDamageScale();
+  [[nodiscard]] const char *UnitName() const override;
+  [[nodiscard]] const char *Author() const override;
+
+  uint32_t fire_count_down_{0};
+  uint32_t Invincible_count_down_{100};
+  uint32_t ChibakuTensei_count_down_{1000};
```
Comment (qinchuanhui) [inline src/battle_game/core/units/assassin_syl.h:25]: Naming problem, shoud look like `fire_count_down_`

**Review by qinchuanhui — APPROVED** [review]
LGTM, you can get the corresponding score

**Pull request closed without merging**